engineering/documentation.md

```diff
@@ -0,0 +1,226 @@
+Examples of these include code coverage requirements, adopted unit testing frameworks,
+and documentation conventions.
+
+A very good example of `CONTRIBUTING.md` is the one from [angular.js](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md).
+
+TL;DR:
+- Audience: Contributors.
+- Content: Steps necessary to modify the project, verify the changes, and contribute back to the project.
+- Example: https://github.com/angular/angular.js/blob/master/README.md
+
+### Reference material
+
+This is the documentation that users will reguarly consult while using your project.
+The actual form depends on whether the project is a library or a tool.
+
+For libraries, this document tends to be autogenerated from source code.
+Tools like godoc, javadoc, etc. provide an easy way to do this.
+For tools, the document will be often hand written, think about `man` pages.
+
+In both cases, this kinds of document tend to grow quite fast as they provide
+detailed information on each feature and the feature's options.
+They do not require a sequential structure. Users often search and read single section.
+Therefore it's recommended to have a good table of contents and search features when possible.
+
+TL;DR:
+- Audience: Users of the project.
+- Content: List of all of the features of the project with full information on arguments, flags, and others.
+- Example: https://docs.angularjs.org/api
+
+### Release Notes
+
+Every new release of your project should be versioned following semantic versioning principles.
+Each new release should come with release notes. These notes are not a list of every single
+PR and issue that have been closed during the release. Instead, the document should explain
+what are the most important changes in the release, such as critical bugs, breaking changes,
+or new features.
+
+TL;DR:
+- Audience: Contributors and users of the project.
+- Content: New features, bug fixes, and other relevant changes since the previous version released.
+- Example: https://golang.org/doc/go1.9
+
+### Guides/Tutorials
+
+For complex tools it is sometimes useful to write guides showing how to perform some specific use case.
+These guides can be short and useful for early beginners, also known as quick starts.
+But often we want to show a complex use case, or show how the project interacts with other technologies.
+In that case we will write a tutorial, or even a codelab if we want the readers to put the concepts to practice.
+
+All of these documents have a goal in common: teach how to do something.
+They follow a step-by-step structure.
+Users often read these guides from top to bottom, unlike reference documentation.
+
+TL;DR:
+- Audience: New users of the project.
+- Content: Detailed guide on how to accomplish something specific using the project.
+- Example: https://docs.angularjs.org/tutorial
+
+### Code of Conduct
+
+All projects under the source{d} organization should mention the Code of Conduct available
```

Comment:
I think it'd be useful here to have a block of markdown you can just copy and paste in a README with the "we have a CoC and it's yada yada yada" and a link to the specific text (https://www.contributor-covenant.org/version/1/4/code-of-conduct.md)

Reply:
done 
also added `.github/CODE_OF_CONDUCT.md` as per https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/